### ISSUE TITLE
[Accessibility]: Screen reader navigation for notebooks

### ISSUE THREAD

**Issue by novoselrok** [sourcegraph/sourcegraph#35180]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34197

### Problem description

On the notebooks list page:

* The star icon in the list items should have an aria label

On the notebook edit page:

* The edit button should announce what it does
* The star svg needs an aria label in the favorite button

### Expected behavior

/

### Assigning labels

- [ ] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [ ] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by sourcegraph-bot-2** [comment]
Heads up @benvenker @lguychard - the "team/search-product" label was applied to this issue.

**Comment by umpox** [comment]
Note:
- The star svg needs an aria label in the favorite button
- The star icon in the list items should have an aria label

These should now be fixed by https://github.com/sourcegraph/sourcegraph/issues/34582. I have updated the description

**Comment by umpox** [comment]
@novoselrok Trying to understand what is left here and if this could just be handed over to GitStart to close the remaining work.

> The edit button should announce what it does

Is that this pencil button? 
<img width="473" alt="image" src="https://user-images.githubusercontent.com/9516420/176161109-80513c5b-e96d-43b8-8750-61fa1ae4c424.png">

**Comment by novoselrok** [comment]
Yep, probably just needs an aria-label. Can be handed over to GitStart, imo.
